File: 397-整数替换.cpp
class Solution {
public:
    int integerReplacement(int n) {
        if(n==1)
            return 0;
        if(n%2==0)
            return integerReplacement(n/2)+1;
        return min(integerReplacement(n/2+1), integerReplacement(n/2))+2;
    }
};

File: 397-整数替换.py
class Solution:
    @cache
    def integerReplacement(self, n: int) -> int:
        if n == 1:
            return 0
        if n%2 == 0:
            return 1 + self.integerReplacement(n // 2)
        return 2 + min(self.integerReplacement(n // 2), self.integerReplacement(n //2 + 1))